paymail_service_provider.go

```diff
@@ -306,3 +317,85 @@ func deriveKey(rawXPubKey string, num uint32) (k *derivedPubKey, err error) {
 	k.pubKey = hex.EncodeToString(k.ecPubKey.SerialiseCompressed())
 	return
 }
+
+func saveBEEFTxInputs(ctx context.Context, c ClientInterface, p2pTx *paymail.P2PTransaction, rts recordIncomingTxStrategy) {
+	if reflect.TypeOf(rts) != reflect.TypeOf(&externalIncomingTx{}) {
+		return
+	}
+
+	inputsToAdd, err := getInputsWhichAreNotInDb(ctx, c, p2pTx)
+	if err != nil {
+		c.Logger().Error(ctx, "error in saveBEEFTxInputs", err)
+	}
+
+	for _, input := range inputsToAdd {
+		err := saveBeefTransactionInput(ctx, c, input)
+		if err != nil {
+			c.Logger().Error(ctx, "error in saveBEEFTxInputs", err)
+		}
+	}
+
+}
+
+func getInputsWhichAreNotInDb(ctx context.Context, c ClientInterface, p2pTx *paymail.P2PTransaction) ([]*bt.Tx, error) {
+	var txIDs []string
+	for _, tx := range p2pTx.DecodedBeef.Transactions {
+		txIDs = append(txIDs, tx.GetTxID())
+	}
+	dbTxs, err := c.GetTransactionsByIDs(ctx, txIDs)
+	if err != nil {
+		return nil, fmt.Errorf("error during getting txs from db: %s", err)
+	}
+
+	txs := make([]*bt.Tx, 0)
+
+	if len(dbTxs) == len(txIDs) {
+		return txs, nil
+	}
+
+	for _, input := range p2pTx.DecodedBeef.Transactions {
+		found := false
+		for _, dbTx := range dbTxs {
+			if dbTx.GetID() == input.GetTxID() {
+				found = true
+				break
+			}
+		}
+		if !found {
+			txs = append(txs, input.Transaction)
+		}
+	}
+
+	return txs, nil
+}
+
+func saveBeefTransactionInput(ctx context.Context, c ClientInterface, input *bt.Tx) error {
```

Comment:
We have to save bump information if exists in BEEF

Reply:
Done